core/corehttp/corehttp.go

```diff
@@ -67,6 +68,17 @@ func listenAndServe(node *core.IpfsNode, addr ma.Multiaddr, handler http.Handler
 		return err
 	}
 
+	host, port, err := net.SplitHostPort(list.Addr().String())
```

Comment:
is there a helper somewhere to construct a multiaddr form a `net.Addr` directly?